### PR TITLE
Investigate typescript upgrade issue

### PR DESCRIPTION
**Pull request by SiegfriedSobkowiak**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [ ] At least one test exists testing the new feature
  - [ ] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [ ] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the clone server
  - [ ] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server
- [x] Make sure that pull requests in the development tools repository, which are connected to this pull requests, 
are merged

### CODE REVIEW

File: dataland-frontend/package.json [hunk 1]
```diff
@@ -47,7 +47,8 @@
     "@vue/eslint-config-prettier": "^9.0.0",
     "@vue/eslint-config-typescript": "^12.0.0",
     "@vue/test-utils": "^2.4.4",
-    "@vue/tsconfig": "^0.1.3",
+    "@vue/tsconfig": "^0.5.1",
+    "@tsconfig/node20": "^20.1.2",
```
Comment (stewal19) [inline dataland-frontend/package.json:51]: Haben wir das vorher nicht gebraucht?

File: dataland-frontend/tsconfig.app.json [hunk 2]
```diff
@@ -13,10 +13,8 @@
       "ESNext",
       "DOM"
     ],
-    // These override some properties in the tsconfig.web.json (and will likely break <script setup> clauses
-    // But are required as otherwise the openApi client won't compile properly
-    "importsNotUsedAsValues": "remove",
-    "preserveValueImports": false,
+    // Setting the following to true breaks the imports of the auto-generated clients
+    "verbatimModuleSyntax": false,
```
Comment (stewal19) [inline dataland-frontend/tsconfig.app.json:17]: Was sind diese Variablen die man auf false setzen muss? (Hier und auch in der tsconfig.commonjs.json Datei)

File: dataland-frontend/tsconfig.cypress-e2e.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 // Config file for cypress E2E tests
 {
-  "extends": "@vue/tsconfig/tsconfig.web.json",
+  "extends": "@vue/tsconfig/tsconfig.dom.json",
```
Comment (stewal19) [inline dataland-frontend/tsconfig.cypress-e2e.json:3]: Hier hat sich nur der Pfad geändert?

Comment (SiegfriedSobkowiak) [reply dataland-frontend/tsconfig.cypress-e2e.json:3]: Ja das hat sich im Rahmen des upgrades geändert.

File: dataland-frontend/tsconfig.app.json (hunk 2 above)
Comment (SiegfriedSobkowiak) [reply dataland-frontend/tsconfig.app.json:17]: Um die hier geht es: "verbatimModuleSyntax": false,
Das ist per default auf true und dann geht alles kaputt

File: dataland-frontend/package.json (hunk 1 above)
Comment (SiegfriedSobkowiak) [reply dataland-frontend/package.json:51]: Ja vorher gab es das nicht. Das node file ist raus und wird durch dieses Paket ersetzt (laut Doku vom vue/tsconfig Paket)

Comment (stewal19) [reply dataland-frontend/package.json:51]: ok

File: dataland-frontend/tsconfig.cypress-e2e.json (hunk 3 above)
Comment (stewal19) [reply dataland-frontend/tsconfig.cypress-e2e.json:3]: ok

File: dataland-frontend/tsconfig.app.json (hunk 2 above)
Comment (stewal19) [reply dataland-frontend/tsconfig.app.json:17]: ok

**Pull request merged**